### PR TITLE
Retry failed jobs three times

### LINKED ISSUE #2488: [builder scheduler] Add re-try logic for job failure
If a job fails to schedule (ie, call to jobsrv does not succeed), or fails a build, the scheduler currently fails the schedule group. This exacts a large penalty when there are a lot of build jobs in the schedule group, as the reason for failure could be something intermittent. This card is for adding re-try logic to the scheduler for job schedule or build failure.

### PR DESCRIPTION
**Pull request by raskchanky**
Pretty self explanatory.

Fixes #2488 

tenor-230250752

Signed-off-by: Josh Black <raskchanky@gmail.com>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-worker/src/runner/mod.rs [hunk 1]
```diff
@@ -457,6 +458,8 @@ pub struct RunnerMgr {
     sock: zmq::Socket,
     msg: zmq::Message,
     config: Arc<RwLock<Config>>,
+    max_retries: u8,
```
Comment (mwrock) [inline components/builder-worker/src/runner/mod.rs]: might be better to put these in the config

Comment (raskchanky) [inline components/builder-worker/src/runner/mod.rs]: Good call.

**Comment by raskchanky** [comment]
There's been some discussion that this might not be the way we want to handle job retries after all.  Let's have that discussion before merging this.

**Comment by reset** [comment]
We're gonna close this one out - per the discussion with @chefsalim and @raskchanky we've got a different approach we'd like to take

**Pull request closed without merging**